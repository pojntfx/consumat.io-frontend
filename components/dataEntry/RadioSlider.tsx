import { ChangeEvent, useEffect, useState } from "react";
import CSS from "csstype";

type RadioSliderProps = {
  name: string;
  value: string;
  options: string[];
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
  className?: string;
};

const RadioSlider = ({
  name,
  value,
  options,
  onChange,
  className,
}: RadioSliderProps) => {
  const [indicator, setIndicator] = useState<CSS.Properties>();
  useEffect(() => {
    setIndicator({
      width: `${100 / options.length}%`,
      left: `${(100 / options.length) * options.indexOf(value)}%`,
    });
  }, [value, options]);

  return (
    <div className={"radioSlider " + className}>
      {options.map((option) => (
        <label key={option} htmlFor={option}>
          <input
            type="radio"
            name={name}
            value={option}
            id={option}
            checked={value === option}
            onChange={(event) => {
              onChange?.(event);
            }}
          />
          <div>{option}</div>
        </label>
      ))}
      <div className="indicator" style={indicator} />
    </div>
  );
};

export default RadioSlider;
