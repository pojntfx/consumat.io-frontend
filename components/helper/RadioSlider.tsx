import { Dispatch, SetStateAction, useEffect, useState } from "react";
import styles from "../../styles/RadioSlider.module.css";

type RadioSliderProps = {
  name: string;
  value: string;
  onChange: Dispatch<SetStateAction<string>>;
  options: string[];
  className?: string;
};

const RadioSlider = ({
  name,
  value,
  onChange,
  options,
  className,
}: RadioSliderProps) => {
  const [indicator, setIndicator] = useState<any>();
  useEffect(() => {
    setIndicator({
      width: `${100 / options.length}%`,
      left: `${(100 / options.length) * options.indexOf(value)}%`,
    });
  }, [value, options]);

  return (
    <div
      className={
        styles.toggleSwitch +
        " " +
        className +
        " bg-gradient-to-br from-white to-white dark:from-gray-600 dark:to-gray-700"
      }
    >
      {options.map((option) => (
        <label key={option} htmlFor={option}>
          <input
            type="radio"
            name={name}
            value={option}
            id={option}
            checked={value === option}
            onChange={(event) => {
              onChange(event.target.value);
            }}
          />
          <div>{option}</div>
        </label>
      ))}
      <div className={styles.indicator} style={indicator} />
    </div>
  );
};

export default RadioSlider;
