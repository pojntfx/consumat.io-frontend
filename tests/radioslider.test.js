import React from "react";
import renderer from "react-test-renderer";
import RadioSlider from "../components/dataEntry/RadioSlider";
import testExamples from "./testExamples";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <RadioSlider
        options={testExamples.exampleOptions}
        value={testExamples.exampleValue}
        name={testExamples.exampleName}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
