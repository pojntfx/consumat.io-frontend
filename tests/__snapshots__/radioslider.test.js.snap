// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="radioSlider undefined"
>
  <label
    htmlFor="sq-AL"
  >
    <input
      checked={false}
      id="sq-AL"
      name="language"
      onChange={[Function]}
      type="radio"
      value="sq-AL"
    />
    <div>
      sq-AL
    </div>
  </label>
  <label
    htmlFor="ar-SA"
  >
    <input
      checked={false}
      id="ar-SA"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ar-SA"
    />
    <div>
      ar-SA
    </div>
  </label>
  <label
    htmlFor="ar-AE"
  >
    <input
      checked={false}
      id="ar-AE"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ar-AE"
    />
    <div>
      ar-AE
    </div>
  </label>
  <label
    htmlFor="eu-ES"
  >
    <input
      checked={false}
      id="eu-ES"
      name="language"
      onChange={[Function]}
      type="radio"
      value="eu-ES"
    />
    <div>
      eu-ES
    </div>
  </label>
  <label
    htmlFor="be-BY"
  >
    <input
      checked={false}
      id="be-BY"
      name="language"
      onChange={[Function]}
      type="radio"
      value="be-BY"
    />
    <div>
      be-BY
    </div>
  </label>
  <label
    htmlFor="bn-BD"
  >
    <input
      checked={false}
      id="bn-BD"
      name="language"
      onChange={[Function]}
      type="radio"
      value="bn-BD"
    />
    <div>
      bn-BD
    </div>
  </label>
  <label
    htmlFor="bg-BG"
  >
    <input
      checked={false}
      id="bg-BG"
      name="language"
      onChange={[Function]}
      type="radio"
      value="bg-BG"
    />
    <div>
      bg-BG
    </div>
  </label>
  <label
    htmlFor="cn-CN"
  >
    <input
      checked={false}
      id="cn-CN"
      name="language"
      onChange={[Function]}
      type="radio"
      value="cn-CN"
    />
    <div>
      cn-CN
    </div>
  </label>
  <label
    htmlFor="ca-ES"
  >
    <input
      checked={false}
      id="ca-ES"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ca-ES"
    />
    <div>
      ca-ES
    </div>
  </label>
  <label
    htmlFor="ch-GU"
  >
    <input
      checked={false}
      id="ch-GU"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ch-GU"
    />
    <div>
      ch-GU
    </div>
  </label>
  <label
    htmlFor="zh-CN"
  >
    <input
      checked={false}
      id="zh-CN"
      name="language"
      onChange={[Function]}
      type="radio"
      value="zh-CN"
    />
    <div>
      zh-CN
    </div>
  </label>
  <label
    htmlFor="zh-HK"
  >
    <input
      checked={false}
      id="zh-HK"
      name="language"
      onChange={[Function]}
      type="radio"
      value="zh-HK"
    />
    <div>
      zh-HK
    </div>
  </label>
  <label
    htmlFor="zh-TW"
  >
    <input
      checked={false}
      id="zh-TW"
      name="language"
      onChange={[Function]}
      type="radio"
      value="zh-TW"
    />
    <div>
      zh-TW
    </div>
  </label>
  <label
    htmlFor="cs-CZ"
  >
    <input
      checked={false}
      id="cs-CZ"
      name="language"
      onChange={[Function]}
      type="radio"
      value="cs-CZ"
    />
    <div>
      cs-CZ
    </div>
  </label>
  <label
    htmlFor="da-DK"
  >
    <input
      checked={false}
      id="da-DK"
      name="language"
      onChange={[Function]}
      type="radio"
      value="da-DK"
    />
    <div>
      da-DK
    </div>
  </label>
  <label
    htmlFor="nl-NL"
  >
    <input
      checked={false}
      id="nl-NL"
      name="language"
      onChange={[Function]}
      type="radio"
      value="nl-NL"
    />
    <div>
      nl-NL
    </div>
  </label>
  <label
    htmlFor="en-AU"
  >
    <input
      checked={false}
      id="en-AU"
      name="language"
      onChange={[Function]}
      type="radio"
      value="en-AU"
    />
    <div>
      en-AU
    </div>
  </label>
  <label
    htmlFor="en-CA"
  >
    <input
      checked={false}
      id="en-CA"
      name="language"
      onChange={[Function]}
      type="radio"
      value="en-CA"
    />
    <div>
      en-CA
    </div>
  </label>
  <label
    htmlFor="en-IE"
  >
    <input
      checked={false}
      id="en-IE"
      name="language"
      onChange={[Function]}
      type="radio"
      value="en-IE"
    />
    <div>
      en-IE
    </div>
  </label>
  <label
    htmlFor="en-NZ"
  >
    <input
      checked={false}
      id="en-NZ"
      name="language"
      onChange={[Function]}
      type="radio"
      value="en-NZ"
    />
    <div>
      en-NZ
    </div>
  </label>
  <label
    htmlFor="en-GB"
  >
    <input
      checked={false}
      id="en-GB"
      name="language"
      onChange={[Function]}
      type="radio"
      value="en-GB"
    />
    <div>
      en-GB
    </div>
  </label>
  <label
    htmlFor="en-US"
  >
    <input
      checked={false}
      id="en-US"
      name="language"
      onChange={[Function]}
      type="radio"
      value="en-US"
    />
    <div>
      en-US
    </div>
  </label>
  <label
    htmlFor="eo-EO"
  >
    <input
      checked={false}
      id="eo-EO"
      name="language"
      onChange={[Function]}
      type="radio"
      value="eo-EO"
    />
    <div>
      eo-EO
    </div>
  </label>
  <label
    htmlFor="et-EE"
  >
    <input
      checked={false}
      id="et-EE"
      name="language"
      onChange={[Function]}
      type="radio"
      value="et-EE"
    />
    <div>
      et-EE
    </div>
  </label>
  <label
    htmlFor="fa-IR"
  >
    <input
      checked={false}
      id="fa-IR"
      name="language"
      onChange={[Function]}
      type="radio"
      value="fa-IR"
    />
    <div>
      fa-IR
    </div>
  </label>
  <label
    htmlFor="fi-FI"
  >
    <input
      checked={false}
      id="fi-FI"
      name="language"
      onChange={[Function]}
      type="radio"
      value="fi-FI"
    />
    <div>
      fi-FI
    </div>
  </label>
  <label
    htmlFor="fr-CA"
  >
    <input
      checked={false}
      id="fr-CA"
      name="language"
      onChange={[Function]}
      type="radio"
      value="fr-CA"
    />
    <div>
      fr-CA
    </div>
  </label>
  <label
    htmlFor="fr-FR"
  >
    <input
      checked={false}
      id="fr-FR"
      name="language"
      onChange={[Function]}
      type="radio"
      value="fr-FR"
    />
    <div>
      fr-FR
    </div>
  </label>
  <label
    htmlFor="gl-ES"
  >
    <input
      checked={false}
      id="gl-ES"
      name="language"
      onChange={[Function]}
      type="radio"
      value="gl-ES"
    />
    <div>
      gl-ES
    </div>
  </label>
  <label
    htmlFor="ka-GE"
  >
    <input
      checked={false}
      id="ka-GE"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ka-GE"
    />
    <div>
      ka-GE
    </div>
  </label>
  <label
    htmlFor="de-AT"
  >
    <input
      checked={false}
      id="de-AT"
      name="language"
      onChange={[Function]}
      type="radio"
      value="de-AT"
    />
    <div>
      de-AT
    </div>
  </label>
  <label
    htmlFor="de-DE"
  >
    <input
      checked={false}
      id="de-DE"
      name="language"
      onChange={[Function]}
      type="radio"
      value="de-DE"
    />
    <div>
      de-DE
    </div>
  </label>
  <label
    htmlFor="de-CH"
  >
    <input
      checked={false}
      id="de-CH"
      name="language"
      onChange={[Function]}
      type="radio"
      value="de-CH"
    />
    <div>
      de-CH
    </div>
  </label>
  <label
    htmlFor="el-GR"
  >
    <input
      checked={false}
      id="el-GR"
      name="language"
      onChange={[Function]}
      type="radio"
      value="el-GR"
    />
    <div>
      el-GR
    </div>
  </label>
  <label
    htmlFor="he-IL"
  >
    <input
      checked={false}
      id="he-IL"
      name="language"
      onChange={[Function]}
      type="radio"
      value="he-IL"
    />
    <div>
      he-IL
    </div>
  </label>
  <label
    htmlFor="hi-IN"
  >
    <input
      checked={false}
      id="hi-IN"
      name="language"
      onChange={[Function]}
      type="radio"
      value="hi-IN"
    />
    <div>
      hi-IN
    </div>
  </label>
  <label
    htmlFor="hu-HU"
  >
    <input
      checked={false}
      id="hu-HU"
      name="language"
      onChange={[Function]}
      type="radio"
      value="hu-HU"
    />
    <div>
      hu-HU
    </div>
  </label>
  <label
    htmlFor="id-ID"
  >
    <input
      checked={false}
      id="id-ID"
      name="language"
      onChange={[Function]}
      type="radio"
      value="id-ID"
    />
    <div>
      id-ID
    </div>
  </label>
  <label
    htmlFor="it-IT"
  >
    <input
      checked={false}
      id="it-IT"
      name="language"
      onChange={[Function]}
      type="radio"
      value="it-IT"
    />
    <div>
      it-IT
    </div>
  </label>
  <label
    htmlFor="ja-JP"
  >
    <input
      checked={false}
      id="ja-JP"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ja-JP"
    />
    <div>
      ja-JP
    </div>
  </label>
  <label
    htmlFor="kn-IN"
  >
    <input
      checked={false}
      id="kn-IN"
      name="language"
      onChange={[Function]}
      type="radio"
      value="kn-IN"
    />
    <div>
      kn-IN
    </div>
  </label>
  <label
    htmlFor="kk-KZ"
  >
    <input
      checked={false}
      id="kk-KZ"
      name="language"
      onChange={[Function]}
      type="radio"
      value="kk-KZ"
    />
    <div>
      kk-KZ
    </div>
  </label>
  <label
    htmlFor="ko-KR"
  >
    <input
      checked={false}
      id="ko-KR"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ko-KR"
    />
    <div>
      ko-KR
    </div>
  </label>
  <label
    htmlFor="lv-LV"
  >
    <input
      checked={false}
      id="lv-LV"
      name="language"
      onChange={[Function]}
      type="radio"
      value="lv-LV"
    />
    <div>
      lv-LV
    </div>
  </label>
  <label
    htmlFor="lt-LT"
  >
    <input
      checked={false}
      id="lt-LT"
      name="language"
      onChange={[Function]}
      type="radio"
      value="lt-LT"
    />
    <div>
      lt-LT
    </div>
  </label>
  <label
    htmlFor="ms-MY"
  >
    <input
      checked={false}
      id="ms-MY"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ms-MY"
    />
    <div>
      ms-MY
    </div>
  </label>
  <label
    htmlFor="ms-SG"
  >
    <input
      checked={false}
      id="ms-SG"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ms-SG"
    />
    <div>
      ms-SG
    </div>
  </label>
  <label
    htmlFor="ml-IN"
  >
    <input
      checked={false}
      id="ml-IN"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ml-IN"
    />
    <div>
      ml-IN
    </div>
  </label>
  <label
    htmlFor="nb-NO"
  >
    <input
      checked={false}
      id="nb-NO"
      name="language"
      onChange={[Function]}
      type="radio"
      value="nb-NO"
    />
    <div>
      nb-NO
    </div>
  </label>
  <label
    htmlFor="no-NO"
  >
    <input
      checked={false}
      id="no-NO"
      name="language"
      onChange={[Function]}
      type="radio"
      value="no-NO"
    />
    <div>
      no-NO
    </div>
  </label>
  <label
    htmlFor="pl-PL"
  >
    <input
      checked={false}
      id="pl-PL"
      name="language"
      onChange={[Function]}
      type="radio"
      value="pl-PL"
    />
    <div>
      pl-PL
    </div>
  </label>
  <label
    htmlFor="pt-BR"
  >
    <input
      checked={true}
      id="pt-BR"
      name="language"
      onChange={[Function]}
      type="radio"
      value="pt-BR"
    />
    <div>
      pt-BR
    </div>
  </label>
  <label
    htmlFor="pt-PT"
  >
    <input
      checked={false}
      id="pt-PT"
      name="language"
      onChange={[Function]}
      type="radio"
      value="pt-PT"
    />
    <div>
      pt-PT
    </div>
  </label>
  <label
    htmlFor="ro-RO"
  >
    <input
      checked={false}
      id="ro-RO"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ro-RO"
    />
    <div>
      ro-RO
    </div>
  </label>
  <label
    htmlFor="ru-RU"
  >
    <input
      checked={false}
      id="ru-RU"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ru-RU"
    />
    <div>
      ru-RU
    </div>
  </label>
  <label
    htmlFor="sr-RS"
  >
    <input
      checked={false}
      id="sr-RS"
      name="language"
      onChange={[Function]}
      type="radio"
      value="sr-RS"
    />
    <div>
      sr-RS
    </div>
  </label>
  <label
    htmlFor="si-LK"
  >
    <input
      checked={false}
      id="si-LK"
      name="language"
      onChange={[Function]}
      type="radio"
      value="si-LK"
    />
    <div>
      si-LK
    </div>
  </label>
  <label
    htmlFor="sk-SK"
  >
    <input
      checked={false}
      id="sk-SK"
      name="language"
      onChange={[Function]}
      type="radio"
      value="sk-SK"
    />
    <div>
      sk-SK
    </div>
  </label>
  <label
    htmlFor="sl-SI"
  >
    <input
      checked={false}
      id="sl-SI"
      name="language"
      onChange={[Function]}
      type="radio"
      value="sl-SI"
    />
    <div>
      sl-SI
    </div>
  </label>
  <label
    htmlFor="es-MX"
  >
    <input
      checked={false}
      id="es-MX"
      name="language"
      onChange={[Function]}
      type="radio"
      value="es-MX"
    />
    <div>
      es-MX
    </div>
  </label>
  <label
    htmlFor="es-ES"
  >
    <input
      checked={false}
      id="es-ES"
      name="language"
      onChange={[Function]}
      type="radio"
      value="es-ES"
    />
    <div>
      es-ES
    </div>
  </label>
  <label
    htmlFor="sv-SE"
  >
    <input
      checked={false}
      id="sv-SE"
      name="language"
      onChange={[Function]}
      type="radio"
      value="sv-SE"
    />
    <div>
      sv-SE
    </div>
  </label>
  <label
    htmlFor="tl-PH"
  >
    <input
      checked={false}
      id="tl-PH"
      name="language"
      onChange={[Function]}
      type="radio"
      value="tl-PH"
    />
    <div>
      tl-PH
    </div>
  </label>
  <label
    htmlFor="ta-IN"
  >
    <input
      checked={false}
      id="ta-IN"
      name="language"
      onChange={[Function]}
      type="radio"
      value="ta-IN"
    />
    <div>
      ta-IN
    </div>
  </label>
  <label
    htmlFor="te-IN"
  >
    <input
      checked={false}
      id="te-IN"
      name="language"
      onChange={[Function]}
      type="radio"
      value="te-IN"
    />
    <div>
      te-IN
    </div>
  </label>
  <label
    htmlFor="th-TH"
  >
    <input
      checked={false}
      id="th-TH"
      name="language"
      onChange={[Function]}
      type="radio"
      value="th-TH"
    />
    <div>
      th-TH
    </div>
  </label>
  <label
    htmlFor="tr-TR"
  >
    <input
      checked={false}
      id="tr-TR"
      name="language"
      onChange={[Function]}
      type="radio"
      value="tr-TR"
    />
    <div>
      tr-TR
    </div>
  </label>
  <label
    htmlFor="uk-UA"
  >
    <input
      checked={false}
      id="uk-UA"
      name="language"
      onChange={[Function]}
      type="radio"
      value="uk-UA"
    />
    <div>
      uk-UA
    </div>
  </label>
  <label
    htmlFor="vi-VN"
  >
    <input
      checked={false}
      id="vi-VN"
      name="language"
      onChange={[Function]}
      type="radio"
      value="vi-VN"
    />
    <div>
      vi-VN
    </div>
  </label>
  <label
    htmlFor="zu-ZA"
  >
    <input
      checked={false}
      id="zu-ZA"
      name="language"
      onChange={[Function]}
      type="radio"
      value="zu-ZA"
    />
    <div>
      zu-ZA
    </div>
  </label>
  <div
    className="indicator"
  />
</div>
`;
